python/lsst/pipe/tasks/make_direct_warp.py

```diff
@@ -65,6 +68,52 @@
 )
 
 
+class TrivialBackgroundList:
+    """Trivial class that quacks like BackgroundList and gives 0 as image."""
+
+    @classmethod
+    def getImage(cls) -> float:
+        return 0.0
```

Comment:
Another option might be to drop this class, allow `process` to accept `None` for its background list arguments regardless of what the config options are, and move the check for consistency between input arguments and configuration up to `run`.

Reply:
I certainly didn't like the way the code read when I had to check if the backgrounds were `None` before calling `getImage`, but I am on board with any of the solutions here. I'll give it some more thought and am happy to just add it on this ticket.

Reply:
Alright, this is what I'm inclining towards:

1. Keep the `TrivialBackgroundList` class, but with a leading underscore name and drop the `getImage` method.
2. Add `apply` and `unapply` methods (names TBD) to `WarpDetectorInputs`, since the operations happen on the components of that dataclass. These would do raise a `RuntimeError` if the background is `None` (current behavior), do nothing if the background is a `_TrivialBackgroundList` and do addition/subtraction only otherwise. It has the advantage of keeping track of whether we are dealing with background subtracted/added image or not.